teuthology/orchestra/remote.py

```diff
@@ -505,7 +505,8 @@ def run(self, **kwargs):
         if not self.ssh or \
            not self.ssh.get_transport() or \
            not self.ssh.get_transport().is_active():
-            self.reconnect()
+            if not self.reconnect():
+                raise Exception(f'Cannot connect to remote host {self.shortname}')
```

Comment:
Should this be 'Cannot reconnect to remote host'?

Reply:
I think it should not, because the self.reconnect() is just an utility function and it is used in general for both initial connect and reconnect after connection got lost and transport is not available. And as in this case when run is executed for the first time it would be weird to see in log that it is going to reconnect during the first time.

Reply:
Makes sense

Reply:
What about raising a [ConnectionLostError](https://github.com/ceph/teuthology/blob/master/teuthology/exceptions.py#L111) here?

Reply:
> What about raising a [ConnectionLostError](https://github.com/ceph/teuthology/blob/master/teuthology/exceptions.py#L111) here?

@zmc see the comment above, it is not about connection lost only, but for establishing initial connection either.

Reply:
I did see that, but if we're going to raise an exception we should give a way for callers to catch that specific exception. I think it should either be `ConnectionLostError` or a sub- or super-class of it. Or we could rename that exception to just `ConnectionError` I suppose

Reply:
@zmc sorry I've merged already, but following your comment, in current code organisation and usage of the run method the differentiation of all cases will cause more just a single line and more hours for testing the code correctness, the goal of this patch is to fix the "silence". Also I am not sure it is really needed for caller of run, to see what exact it error it was, start connection or lost connection or what ever connection, what is the benefit?